src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -58544,6 +58544,231 @@ void Test1(Test? x1)
+class C
+{
+    event System.Action? E1;
+    void M()
+    {
+        E1 -= () => { };
+        E1.Invoke(); // 1
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (8,9): warning CS8602: Dereference of a possibly null reference.
+                //         E1.Invoke(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "E1").WithLocation(8, 9));
+        }
+
+        [Fact, WorkItem(42557, "https://github.com/dotnet/roslyn/issues/42557")]
+        public void EventSubtraction_MaybeNullArg_NullableEvent()
+        {
+            var source = @"
+class C
+{
+    event System.Action? E1;
+    void M(System.Action? e2)
+    {
+        E1 -= e2;
+        E1.Invoke(); // 1
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (8,9): warning CS8602: Dereference of a possibly null reference.
+                //         E1.Invoke(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "E1").WithLocation(8, 9));
+        }
+
+        [Fact, WorkItem(42557, "https://github.com/dotnet/roslyn/issues/42557")]
+        public void EventSubtraction_MaybeNullArg_NonNullableEvent()
+        {
+            var source = @"
+class C
+{
+    event System.Action E1 = null!;
+    void M(System.Action? e2)
+    {
+        E1 -= e2;
+        E1.Invoke(); // 1
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (8,9): warning CS8602: Dereference of a possibly null reference.
+                //         E1.Invoke(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "E1").WithLocation(8, 9));
+        }
+
+        [Fact, WorkItem(42557, "https://github.com/dotnet/roslyn/issues/42557")]
+        public void EventSubtraction_NullArg_NullableEvent()
+        {
+            var source = @"
+class C
+{
+    event System.Action? E1;
+    void M()
+    {
+        E1 -= null;
+        E1.Invoke(); // 1
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (8,9): warning CS8602: Dereference of a possibly null reference.
+                //         E1.Invoke(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "E1").WithLocation(8, 9));
+        }
+
+        [Fact, WorkItem(42557, "https://github.com/dotnet/roslyn/issues/42557")]
+        public void EventSubtraction_NullArg_NonNullableEvent()
+        {
+            var source = @"
+class C
+{
+    event System.Action E1 = null!;
+    void M()
+    {
+        E1 -= null;
+        E1.Invoke();
+    }
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics();
+        }
```

Comment:
>} [](start = 8, length = 1)

Consider combining tests where reasonable, to reduce the number of similar tests and make it easier to see the differences. For instance, consider combining add/remove pairs:
```
class C
{
    event System.Action E1 = null!;
    void M1()
    {
        E1 += () => { };
        E1.Invoke();
    }
    void M2()
    {
        E1 -= () => { };
        E1.Invoke(); // 1
    }
}
```